packages/common/src/graphql/entities.ts

```diff
@@ -1,14 +1,126 @@
 // Copyright 2020-2021 OnFinality Limited authors & contributors
 // SPDX-License-Identifier: Apache-2.0
 
-import {GraphQLSchema, GraphQLObjectType, isObjectType} from 'graphql';
+import {
+  GraphQLSchema,
+  GraphQLObjectType,
+  isObjectType,
+  isNonNullType,
+  getNullableType,
+  isListType,
+  assertListType, getDirectiveValues, GraphQLField
+} from 'graphql';
 import {DirectiveName} from './constant';
 
+import {
+  GraphQLModelsRelations,
+  GraphQLModelsType,
+  GraphQLEntityField,
+  FieldScalar,
+  GraphQLRelationsType
+} from './types';
+
 export function getAllEntities(schema: GraphQLSchema): GraphQLObjectType[] {
   return Object.entries(schema.getTypeMap())
     .filter(([, node]) => node.astNode?.directives?.find(({name: {value}}) => value === DirectiveName.Entity))
     .map(([, node]) => node)
     .filter(isObjectType);
 }
 
-// TODO: GraphQLNamedType -> EntitySchema
+export function getAllEntitiesRelations(schema: GraphQLSchema): GraphQLModelsRelations{
+  const modelRelations = {models:[],relations:[]} as GraphQLModelsRelations;
+  const entityNameSet = [];
+  const reverseLookupRelations = [];
+
+  const derivedFrom = schema.getDirective('derivedFrom');
+  const entities = Object.entries(schema.getTypeMap())
+      .filter(([, node]) => node.astNode?.directives?.find(({name: {value}}) => value === DirectiveName.Entity))
+      .map(([, node]) => node)
+      .filter(isObjectType);
+
+  for (const entity of entities){
+    entityNameSet.push(entity.name);
+  }
+
+  for (const entity of entities){
+    const newModel: GraphQLModelsType=  {
+      name: entity.name,
+      fields : []
+    }
+    const fields = entity.getFields();
+    for (const k in fields){
```

Comment:
if you change from 'for..in' to 'for..of', the code will be much simpler here, no hashOwnProperty check needed.